swift/Arrow/Sources/Arrow/ArrowCExporter.swift

```diff
@@ -0,0 +1,100 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import ArrowC
+
+extension String {
+    var cstring: UnsafePointer<CChar> {
+        (self as NSString).cString(using: String.Encoding.utf8.rawValue)!
+    }
+}
+
+public class ArrowCExporter {
+    private class ExportData {
+        let id: String
+        var cArray = ArrowC.ArrowArray()
+        private let arrowData: ArrowData
+        private(set) var data = [UnsafeRawPointer?]()
+        private(set) var buffers: UnsafeMutablePointer<UnsafeRawPointer?>
+        init(_ arrowData: ArrowData) {
+            id = UUID().uuidString
+            // keep a reference to the ArrowData
+            // obj so the memory doesn't get
+            // deallocated
+            self.arrowData = arrowData
+            for arrowBuffer in arrowData.buffers {
+                data.append(arrowBuffer.rawPointer)
+            }
+
+            self.buffers = UnsafeMutablePointer(mutating: data)
+            ArrowCExporter.exportedData[id.hashValue] = self
```

Comment:
I'm curious: why do you generate a string UUID and then use its hash value as index, instead of directly generating a random integer?
Or, perhaps even safer, generate an [atomically-incremented integer](https://www.swift.org/blog/swift-atomics/)?

Reply:
I had started with UUID but it devolved into the hashvalue due to the private_data issue you noticed in another comment.  I will update to use the atomics as the UUID implementation isn't checking if the hashValue has been used before and looking for a free one.  Thank you!